src/Init/Meta.lean

```diff
@@ -234,6 +235,47 @@ instance monadNameGeneratorLift (m n : Type → Type) [MonadLift m n] [MonadName
   setNGen := fun ngen => liftM (setNGen ngen : m _)
 }
 
+namespace TSyntax
+
+instance : Coe (TSyntax [k]) (TSyntax (k :: ks)) where
+  coe stx := ⟨stx⟩
+
+instance [Coe (TSyntax [k]) (TSyntax ks)] : Coe (TSyntax [k]) (TSyntax (k' :: ks)) where
+  coe stx := ⟨stx⟩
+
+instance : Coe (TSyntax identKind) (TSyntax `term) where
+  coe s := ⟨s.raw⟩
+
+instance : CoeDep (TSyntax `term) ⟨Syntax.ident info ss n res⟩ (TSyntax `ident) where
+  coe := ⟨Syntax.ident info ss n res⟩
+
+instance : Coe (TSyntax strLitKind) (TSyntax `term) where
+  coe s := ⟨s.raw⟩
+
+instance : Coe (TSyntax nameLitKind) (TSyntax `term) where
+  coe s := ⟨s.raw⟩
+
+instance : Coe (TSyntax numLitKind) (TSyntax `term) where
+  coe s := ⟨s.raw⟩
+
+instance : Coe (TSyntax charLitKind) (TSyntax `term) where
+  coe s := ⟨s.raw⟩
+
+instance : Coe (TSyntax numLitKind) (TSyntax `prec) where
+  coe s := ⟨s.raw⟩
```

Comment:
We could make these instances "safe" by moving `Unhygienic` into here so that we can do e.g. ``Unhygienic.run `(prec| $s:num)``

Reply:
Could we add a command that automates this construction (presumably with some unconventional parser use)?
```lean
tsyntax_coe ident : declId

-- "expands" to
instance : Coe (TSyntax `ident) (TSyntax ``declId) where
  coe s := Unhygienic.run `(declId|$s:ident)
```

Reply:
Yes, I was deliberating/dreading that idea... :)